python/adapters.h

```diff
@@ -21,21 +21,21 @@ typedef std::unordered_set<Embag::RosValue::Type, EnumClassHash> RosValueTypeSet
 typedef std::unordered_set<Embag::RosValue::Type> RosValueTypeSet;
 #endif
 
-py::dict rosValueToDict(const Embag::RosValue::Pointer &ros_value, const RosValueTypeSet &types_to_unpack);
-py::list rosValueToList(const Embag::RosValue::Pointer &ros_value, const RosValueTypeSet &types_to_unpack);
-py::object castValue(const Embag::RosValue::Pointer &value);
+py::dict rosValueToDict(const Embag::RosValue::Pointer &ros_value, const RosValueTypeSet &types_to_unpack, py::object ros_time_py_type);
+py::list rosValueToList(const Embag::RosValue::Pointer &ros_value, const RosValueTypeSet &types_to_unpack, py::object ros_time_py_type);
+py::object castValue(const Embag::RosValue::Pointer &value, py::object ros_time_py_type);
 
 // By default, it doesn't make sense to return a memoryview for python non-primitive types
 const RosValueTypeSet default_types_to_unpack = {
   Embag::RosValue::Type::ros_duration,
   Embag::RosValue::Type::ros_time
 };
 
-py::object primitiveArrayToPyObject(const Embag::RosValue::Pointer &primitive_array, const RosValueTypeSet &types_to_unpack=default_types_to_unpack) {
+py::object primitiveArrayToPyObject(const Embag::RosValue::Pointer &primitive_array, const RosValueTypeSet &types_to_unpack=default_types_to_unpack, py::object ros_time_py_type=py::none()) {
```

Comment:
NIT: These lines seem to be too big. Can we re-format ?

Reply:
Sure thing